libqtile/state.py

```diff
@@ -31,10 +32,17 @@ def __init__(self, qtile):
         # configurations.
         self.groups = {}
         self.screens = {}
+        self.focusHistory = {}
         self.current_screen = 0
+        self.layoutMap = {}
 
         for group in qtile.groups:
             self.groups[group.name] = group.layout.name
+            self.layoutMap[group.name] = {}
+            for layout in group.layouts:
+                self.layoutMap[group.name][layout.name] = layout
+                layout.group = None
```

Comment:
Class `group` has multiple attributes that refer back to `layout` class and it ends up in recursion.If this attribute is not set to `None`, an endless recursion will be caused. Thus for python pickle to work I had to break this recursive chain and I set this attribute to None.Since the mapping stores the `group.name` , I use this to restore this value after restart. In the `apply` method I restore this attribute in `line 64`.